stdlib/public/core/SwiftNativeNSArray.swift

```diff
@@ -263,7 +263,13 @@ extension __SwiftNativeNSArrayWithContiguousStorage: _NSArrayCore {
                                count: Int) {
     let range = range.location ..< range.location + range.length
     let buf = UnsafeBufferPointer(start: objects, count: count)
-    contents.replaceSubrange(range, with: buf)
+    if range == contents.startIndex..<contents.endIndex {
+      contents = Array(buf)
+    } else {
+      // We make an Array here to make sure that something is holding onto the
+      // objects in `buf`, since replaceSubrange could release them
```

Comment:
Oof. Optimize later to check, I guess?

Reply:
If it looks relevant in practice, sure

Reply:
You're performing an extra copy this way, at least formally.

Reply:
(which is the point, of course)

Reply:
Right, but that just brings us down to NSMutableArray's speed, in theory. Anything better here would be a win on top of that which (probably) makes it nice to have rather than necessary.